### PR TITLE
Correct health status of logging with LOG_DISARMED/multiple backends

### LINKED ISSUE #6397: BAD LOGGING prearm fail when logging to file and mavlink
#### Issue details
When the `LOG_BACKEND_TYPE`=3 (file and mavlink), and `LOG_DISARMED`=1 (enabled), there is a "BAD LOGGING" prearm failure. However this seems to be a false error, because logging is working fine.

If you change the backend type to only to file or only to mavlink, there is no failure. Only when trying to do both. And only when disarmed.

This is not just on mine. Everyone running master with this configuration is having the problem for months. I've worked around it by disabling the logging prearm check. But that's not a good fix.

Suggestions on what may be causing this?

#### Version
3.5+

#### Platform
[  ] All
[  ] AntennaTracker
[X] Copter
[  ] Plane
[  ] Rover

#### Airframe type
N/A

#### Hardware type
Pixhawk 2.1 and IMX companion computer

### PR DESCRIPTION
**Pull request by peterbarker**
Closes #6397

### CODE REVIEW

**Comment by Pedals2Paddles** [comment]
I looks like the example files need to be updated too.

File: AntennaTracker/Log.cpp [hunk 1]
```diff
@@ -90,10 +90,8 @@ void Tracker::start_logging()
     if (g.log_bitmask != 0) {
         if (!logging_started) {
```
Comment (peterbarker) [inline AntennaTracker/Log.cpp]: Gah.  Yes.

**Comment by peterbarker** [comment]
@OXINARF  dataflash_block and dataflash_sitl are not used anywhere ATM, 'though you can use dataflash_sitl if you really work at it.

**Comment by peterbarker** [comment]
@OXINARF I think your points have been addressed now - thanks!

**Comment by Pedals2Paddles** [comment]
I loaded this onto my Solo and it works great. No more BAD LOGGING prearm failure.  And logging is working to both file and mavlink. I was able to set the prearm checks back to all, whereas before I had to have the logging prearm check disabled.  Nice job Peter!

@rmackay9, can this get backported into RC8?  Since it is a bigfix, I think it's pretty important to get into 3.5. Any vehicle using a companion computer will greatly appreciate this fix!

**Comment by OXINARF** [comment]
Merged, thanks!

> dataflash_block and dataflash_sitl are not used anywhere ATM, 'though you can use dataflash_sitl if you really work at it.

@peterbarker Any reason to not remove them?

**Comment by peterbarker** [comment]
@OXINARF I'm in two minds about them.  We don't usually carry dead code, and for that reason I think removal wouldn't necessarily be a bad thing.  The maintenance load isn't high.

But why keep them at all?  The ability to do some logging to systems which can't use an SD card or communicate with a companion computer but also have vast amounts of flash?  Seems a thin excuse, really...

**Comment by OXINARF** [comment]
> Seems a thin excuse, really...

@peterbarker A really thin excuse I would say. In any case I think that at least the SITL one can be removed, I'm not seeing anyone using SITL in a system as you described.

**Comment by peterbarker** [comment]
On Tue, 13 Jun 2017, Francisco Ferreira wrote:

>       Seems a thin excuse, really...
> 
> @peterbarker A really thin excuse I would say. In any case I think that at
> least the SITL one can be removed, I'm not seeing anyone using SITL in a
> system as you described.

Ah, but SITL is the only way to test DataFlash_Block :-)

I'll remove them in a different PR.  tridge has already said we could get 
rid of them if they become a nuisance.

**Pull request merged**